File: API/src/main/java/com/insane/apiwtb/config/AppConfig.java
package com.insane.apiwtb.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Configuration;

@Configuration
public class AppConfig {

    @Value("${HOME}")
    private String homePath;

    @Value("${config.mediaFolderName}")
    private String imageFolderName;

    @Value("${config.hostName}")
    private String host;

    private final String imageURL;

    public AppConfig() {
        // End point set on ImageController to get images from server
        // The string mentioned here must be the same as on ImageController endPoint
        this.imageURL = "/api/wtb-v1/image";
    }

    public String getImageURL(String filename) {
        return host+imageURL+"/"+filename;
    }

    public String getImagePath() {
        return this.homePath+"/"+imageFolderName;
    }

    public String getImagePath(String filename) {
        return this.homePath+"/"+imageFolderName+"/"+filename;
    }
}

File: Mobile/Android/app/src/main/java/com/insane/whatsthebra/config/AppConfig.kt
package com.insane.whatsthebra.config

object AppConfig {

    /**
     * API base - host and end-points
     */
    object API {

        private const val hostBase: String = "https://insane-wtb.rcastrucci.com/"
        private const val api: String = "api/wtb-v1/"
        private const val imageEndPointId: String = "image/id/"
        private const val imageEndPoint: String = "image/"

        fun getHostBase(): String {
            return hostBase;
        }

        fun getHost(): String {
            return "${hostBase}${api}";
        }

        fun getImageUrl(imageId: Int): String {
            return "${hostBase}${api}${imageEndPointId}${imageId}"
        }

        fun getImageUrl(imageName: String): String {
            return "${hostBase}${api}${imageEndPoint}${imageName}"
        }

    }

}